blockchair/adapter.js

```diff
@@ -13,25 +14,31 @@ const convertBlockchain = {
   LTC: 'litecoin',
 }
 
+const convertQ = {
+  height: 'blocks',
+}
+
 const execute = (input, callback) => {
   const validator = new Validator(input, customParams)
   if (validator.error) return callback(validator.error.statusCode, validator.errored)
 
   const jobRunID = validator.validated.id
   let blockchain = validator.validated.data.blockchain
+  let q = validator.validated.data.q || 'difficulty'
```

Comment:
We currently use an `endpoint` input param to differentiate between different data points we want to fetch from the external data provider.

I would suggest we don't introduce new behavior to the adapter API at this point but continue using what we have now.

Take a look at how we handle endpoint queries in different providers, and please try to mimic the interface and behavior here.

---

In the future, we are looking to streamline the adapter API. The first thing we should do is list the data points we support and codify them. Here you are adding support for a new data point, let's call it `BlockNumber`. All of our adapters that know how to answer requests for that specific data point should follow the same API (input/output).

In this context, it may make sense to wrap the `json-rpc` adapter into `bitcoin-json-rpc` adapter that will then expose this specific data point (and others).